exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -251,7 +251,7 @@ func (prwe *prwExporter) execute(ctx context.Context, writeReq *prompb.WriteRequ
 	}
 	body, err := io.ReadAll(io.LimitReader(resp.Body, 256))
 	rerr := fmt.Errorf("remote write returned HTTP status %v; err = %w: %s", resp.Status, err, body)
-	if resp.StatusCode >= 500 && resp.StatusCode < 600 {
+	if resp.StatusCode == 429 || resp.StatusCode >= 500 && resp.StatusCode < 600 {
 		return rerr
```

Comment:
Are we actually retrying them? Looks like they are eventually wrapped as permanent https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/prometheusremotewriteexporter/exporter.go#L205

Reply:
In fact, I'm afraid that you right, the param `retry_on_failure` seems useless.